patches/tModLoader/Terraria/ModLoader/Engine/TerrariaSteamClient.cs

```diff
@@ -140,6 +141,11 @@ public enum LaunchResult
 					return;
 				}
 
+				int TerrariaBuildID = SteamApps.GetAppBuildId();
+				Logger.Info("Terraria BuildID: " + TerrariaBuildID);
+				if (TerrariaBuildID < LatestTerrariaBuildID) 
```

Comment:
can you pipe this back to the main process via an error code? This should be the primary point of failure.